test/cql-pytest/test_type_time.py

```diff
@@ -49,3 +57,67 @@ def test_type_time_from_string_prepared(cql, table1):
     for t in ["08:12:54", "08:12:54.123", "08:12:54.123456", "08:12:54.123456789"]:
         cql.execute(stmt, [p, t])
         assert list(cql.execute(f"SELECT t from {table1} where p = {p}")) == [(Time(t),)]
+
+# According to the CQL binary protocol spec, `time` values must be between 0 and 24 hours.
+# Test that Scylla rejects integer values outside of this range
+# https://github.com/apache/cassandra/blob/f5df4b219e063cb24b9cc0c22b6e614506b8d903/doc/native_protocol_v4.spec#L941
+# An 8 byte two's complement long representing nanoseconds since midnight.
+# Valid values are in the range 0 to 86399999999999
+@pytest.mark.xfail(reason="issue #14667")
+def test_invalid_time_int_values(cql, table1):
+    p = unique_key_int()
+    insert_t = cql.prepare(f"INSERT INTO {table1} (p, t) VALUES ({p}, ?)")
+    update_t = cql.prepare(f"UPDATE {table1} SET t = ? WHERE p = {p}")
+
+    # Values within the range are accepted
+    cql.execute(insert_t, (Time(0),))
+    cql.execute(update_t, (Time(123456),))
+    cql.execute(insert_t, (Time(86399999999999),))
+
+    # Values outside of the range are rejected
+    with pytest.raises(InvalidRequest):
```

Comment:
idea: you can do a `for i in [-1, 86400000000000, -1234, ...]` to make this test shorter (and also easily check every one of these times with both insert and update, not just some of them.

Reply:
I dislike tests that have a list of values, because when something fails I have to read the code to find out which value caused the error.
Having each test case in separate case produces error messages that are very clear.